docs/02.API-REFERENCE.md

```diff
@@ -68,6 +68,18 @@ Option bits for [jerry_parse](#jerry_parse) and
  - JERRY_PARSE_NO_OPTS - no options passed
  - JERRY_PARSE_STRICT_MODE - enable strict mode
 
+## jerry_gc_mode_t
+
+Set garbage collection operational mode
```

Comment:
I'm no native English speaker but isn't this "operation mode" or "operating mode"?

Reply:
According to google, `operation mode` is used 262M sites, while `operational mode` 117M sites. Both are used by high profile (e.g. microsoft or google technical docs) sites. I think both are ok, so I don't want to change unless you insist on it.

Reply:
I checked Cambridge dictionary:

operation - noun
operational - adjective
operating - present participle

I would prefer `operational` here.

Reply:
Google: The present participle is a participle that ends in ing. It can be used with the auxilliary verb 'to be' to form the continuous tense. It always takes the 'ing' form of the verb, even irregular verbs have an '...ing' form, in fact virtually all English words that end with 'ing' are present participles.

I think `operating mode` is also correct, if we emphasize how garbage collection is performed. The `operational mode` simply emphasize the behavior.

Reply:
E.g:
operating system - a system which is running (emphasizing continuity)
operational system - a system which is capable of running (not relevant if it is doing it now)
operation system - double noun, usually a phrase or not used at all. I think the latter is true here.

Reply:
As you've said, saying something is 'operational' means it is capable of running. Saying that the mode is operational means that the thing that is running will be able to run in that mode, but also implies that a 'not operational' mode exists, which 'won't be able to run'. This is not the case, as the mode here means a specific a way of operation, and not whether the GC can actually run, or is running.

I'm not sure what behavior you'd like to emphasize by using `operational mode`, yet it feels instinctively wrong to me, and I'd say the suggestions made by @akosthekiss would be the the correct way to go.

Reply:
I didn't exactly say that. Tricky way of turning my words :)

Adjectives extends the meaning of words, while nouns do not. Think about the difference of "wooden house" and "tree house". The former uses an adjective to extend a noun, while the latter is simply a phrase. 

I found "operational mode" in a well known English dictionary:
https://en.oxforddictionaries.com/definition/standby
An operational mode of an electrical appliance in which the power is switched on but the appliance is not actually functioning.

I could not find the same thing with "operation mode".

Reply:
I've consulted an English teacher and operation was preferred over operational. But still no native speaker chiming in on either side. As this is no API function name but a doc text only, I wont be pushing further.